addons/binding/org.openhab.binding.http/src/main/java/org/openhab/binding/http/handler/HttpThingHandler.java

```diff
@@ -0,0 +1,405 @@
+        } catch (final IllegalArgumentException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    public void handleCommand(final ChannelUID channelUID, final Command command) {
+        if (!CHANNEL_STATE.equals(channelUID.getId())) {
+            logger.warn("[{}] unknown channel '{}'", getThing().getUID(), channelUID.getId());
+        } else if (command.equals(RefreshType.REFRESH)) {
+            this.stateRequest.ifPresent(this::fetchState);
+        } else if (!this.commandRequest.isPresent()) {
+            logger.warn("[{}] got command on channel '{}', but no command URL set", getThing().getUID(), channelUID.getId());
+        } else {
+            final CommandRequest commandRequest = this.commandRequest.get();
+            final CommandRequest.Method method = commandRequest.getMethod();
+            final String commandStr = command.toFullString();
+            try {
+                final String transformedCommand = doTransform(commandRequest.getRequestTransform(), commandStr);
+                final URL transformedUrl = formatUrl(commandRequest.getUrl(), transformedCommand);
+                makeHttpRequest(method, transformedUrl, commandRequest.getContentType(), Optional.empty(), Optional.of(commandStr)).whenComplete((response, t) -> {
+                    if (t != null) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Connetion to server failed when sending command: " + t.getMessage());
+                    } else if (response.getResponse().getStatus() / 100 != 2) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Server returned HTTP status " + response.getResponse().getStatus() + " when sending command");
+                    } else {
+                        updateStatus(ThingStatus.ONLINE);
+                        updateState(stateFromResponse(response, commandRequest.getResponseTransform()));
+                    }
+                });
+            } catch (final IllegalArgumentException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    protected void updateStatus(final ThingStatus status, final ThingStatusDetail statusDetail, @Nullable final String description) {
+        final ThingStatusInfo curStatusInfo = getThing().getStatusInfo();
+        if (!Objects.equals(status, curStatusInfo.getStatus()) || !Objects.equals(statusDetail, curStatusInfo.getStatusDetail()) || !Objects.equals(description, curStatusInfo.getDescription())) {
+            super.updateStatus(status, statusDetail, description);
+        }
+    }
+
+    @Override
+    public void handleConfigurationUpdate(final Map<String, Object> configurationParameters) {
+        super.handleConfigurationUpdate(configurationParameters);
```

Comment:
I recommend not to handle this yourself. Have a look at the super implementation. The Handler will be disposed/initialized if the user changes a configuration. And that is perfect for your case, you just parse the config once, in initialize.